### DISCUSSION TITLE
I'm confused about using this SDK with for an application used by my multiple tenants

### DISCUSSION THREAD

**Discussion by awartoft** [microsoftgraph/msgraph-sdk-go#581] [Q&A]
Hello,

I'm part of a team working on a Microsoft teams application and I'm rather confused about how one would go about using this SDK when we are trying to send notifications to users outside of our own tenant as they will be installing the app into their own Microsoft Teams instances. 

From what I have figured from this post
https://techcommunity.microsoft.com/t5/microsoft-graph/teams-activity-feed-notifications-multi-tenant/m-p/3264792

Is that we should be fetching an access token using the following endpoint using the this endpoint
https://learn.microsoft.com/en-us/graph/auth-v2-service?tabs=http#4-request-an-access-token

But how we should either incorporate that endpoint or use it with the SDK I'm totally lost on. 

Would love to get some advice on how the SDK is supposed to be used.

**Comment by baywet** [comment]
I think [this is what you're looking for](https://learn.microsoft.com/en-us/graph/sdks/choose-authentication-providers?tabs=go#client-credentials-provider). The trick being you'll need to instantiate a credential and therefore a new client for each tenant you're trying to issue requests against.
